pinot-azure-filesystem/src/main/java/org/apache/pinot/filesystem/AzurePinotFS.java

```diff
@@ -100,9 +100,28 @@ public boolean delete(URI segmentUri, boolean forceDelete)
   @Override
   public boolean move(URI srcUri, URI dstUri, boolean overwrite)
       throws IOException {
-    if (exists(dstUri) && !overwrite) {
+    if (!exists(srcUri)) {
+      LOGGER.warn("Source {} does not exist", srcUri);
       return false;
     }
+    // if dst is an existing file
+    if (exists(dstUri) && !isDirectory(dstUri)) {
+      if (isDirectory(srcUri)) {
+        String errorMsg = String.format("Source %s is a directory while destination %s is a file", srcUri, dstUri);
+        LOGGER.warn(errorMsg);
+        throw new IOException(errorMsg);
+      } else {
+        if (overwrite) {
+          delete(dstUri, true);
+        } else {
+          // dst file exists, returning
+          return false;
+        }
+      }
+    }
+    // ensures the parent path of dst exists.
+    URI parentUri = Paths.get(dstUri).toUri();
+    _adlStoreClient.createDirectory(parentUri.getPath());
     //rename the file
```

Comment:
This comment is misleading as its assuming a 'file' whereas all of the above code seems more flexible.

Reply:
Adjusted the comment.